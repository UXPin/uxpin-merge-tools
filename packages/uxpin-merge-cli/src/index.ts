import 'source-map-support/register';
export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';
export { buildDesignSystemWithStorybook } from './steps/building/buildDesignSystemWithStorybook';
