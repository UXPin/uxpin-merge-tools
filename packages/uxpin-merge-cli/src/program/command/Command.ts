export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  DELETE_VERSION = 'delete-version',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND: Command = Command.EXPERIMENT;
