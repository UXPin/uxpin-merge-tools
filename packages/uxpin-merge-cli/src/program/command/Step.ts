import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun: boolean;
  exec: (infos: DSMetadata) => any;
}

export type StepExecutor = (designSystem: DSMetadata) => Promise<DSMetadata>;

export type StepWithoutDSExecutor = () => Promise<void>;
