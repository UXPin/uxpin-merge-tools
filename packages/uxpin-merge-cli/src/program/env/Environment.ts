export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  PRODUCTION = 'production',
}

export const DEFAULT_ENVIRONMENT: Environment = Environment.PRODUCTION;
