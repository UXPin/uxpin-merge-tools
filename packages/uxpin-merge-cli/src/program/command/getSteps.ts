import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getDeleteVersionCommandSteps } from './delete_version/getDeleteVersionCommandSteps';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getExperimentationCommandSteps } from './experimentation/getExperimentationCommandSteps';
import { getGeneratePresetsCommandSteps } from './generate_presets/getGeneratePresetsCommandSteps';
import { getInitCommandSteps } from './init/getInitCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args: ProgramArgs): Step[] {
  switch (args.command) {
    case Command.DUMP:
      return getDumpCommandSteps();

    case Command.EXPERIMENT:
      return getExperimentationCommandSteps(args);

    case Command.INIT:
      return getInitCommandSteps(args);

    case Command.GENERATE_PRESETS:
      return getGeneratePresetsCommandSteps(args);

    case Command.PUSH:
      return getPushCommandSteps(args);

    case Command.DELETE_VERSION:
      return getDeleteVersionCommandSteps(args);

    case Command.SERVER:
      return getServerCommandSteps(args);

    case Command.SUMMARY:
      return getSummaryCommandSteps();

    default:
      return [];
  }
}
