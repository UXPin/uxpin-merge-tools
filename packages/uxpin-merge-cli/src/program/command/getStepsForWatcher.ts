import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args: ProgramArgs): Step[] {
  if (args.command === Command.EXPERIMENT) {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
