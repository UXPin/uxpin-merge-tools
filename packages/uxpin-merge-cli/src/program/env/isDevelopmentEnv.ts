import { Environment } from './Environment';

export function isDevelopmentEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.DEVELOPMENT;
}
