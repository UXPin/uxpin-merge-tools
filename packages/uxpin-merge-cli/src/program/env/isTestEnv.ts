import { Environment } from './Environment';

export function isTestEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.TEST;
}
