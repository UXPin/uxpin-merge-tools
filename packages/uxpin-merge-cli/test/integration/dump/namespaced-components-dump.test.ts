import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { emptyLatestCommitStub } from '../../resources/stubs/emptyLatestCommit';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';

const CURRENT_TIMEOUT = 60000;
setTimeoutBeforeAll(CURRENT_TIMEOUT);

jest.mock('../../../src/program/utils/version/getToolVersion');

describe('The dump command', () => {
  const { getTlsPort } = setupStubbyServer(emptyLatestCommitStub);

  describe('run for the namespacedComponents repository', () => {
    it('prints the JSON describing the full repository', async () => {
      // when
      const output: string = await runUXPinMergeCommand({
        cwd: 'resources/designSystems/namespacedComponents',
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [Command.DUMP, '--webpack-config "./webpack.config.js"'],
      });

      // then
      expect(JSON.parse(output)).toMatchSnapshot({
        vcs: expect.objectContaining({
          branchName: expect.any(String),
          commitHash: expect.stringMatching(/[a-z0-9]+/),
        }),
      });
    });
  });

  describe('run for the namespacedComponents repository with invalid components', () => {
    it('prints an error when namespace is duplicated for two components with the same name', async () => {
      expect.assertions(1);

      try {
        // when
        await runUXPinMergeCommand({
          cwd: 'resources/designSystems/namespacedComponents',
          env: {
            UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
            UXPIN_ENV: Environment.TEST,
          },
          params: [Command.DUMP, '--webpack-config "./webpack.config.js"', '--config "./uxpin.invalid.config.js"'],
        });
      } catch (error) {
        // then
        expect(error.stderr).toMatch(/Component \"Card\.Header\.Button\" already exists/);
      }
    });

    it('prints an error when component uses not existing namespace', async () => {
      expect.assertions(1);

      try {
        // when
        await runUXPinMergeCommand({
          cwd: 'resources/designSystems/namespacedComponents',
          env: {
            UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
            UXPIN_ENV: Environment.TEST,
          },
          params: [Command.DUMP, '--webpack-config "./webpack.config.js"', '--config "./uxpin.invalid2.config.js"'],
        });
      } catch (error) {
        // then
        expect(error.stderr).toMatch(/Namespace \"Some\" does not exist/);
      }
    });
  });
});
