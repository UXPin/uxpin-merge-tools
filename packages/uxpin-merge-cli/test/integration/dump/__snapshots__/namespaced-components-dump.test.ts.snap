// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The dump command run for the namespacedComponents repository prints the JSON describing the full repository 1`] = `
Object {
  "categorizedComponents": Array [
    Object {
      "components": Array [
        Object {
          "defaultExported": true,
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Card",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Card/Card.tsx",
            },
            "presets": Array [
              Object {
                "path": "src/components/Card/presets/0-with direct imports.jsx",
              },
              Object {
                "path": "src/components/Card/presets/1-using namespaces.jsx",
              },
            ],
          },
          "name": "Card",
          "presets": Array [
            Object {
              "elements": Object {
                "1": Object {
                  "name": "Card",
                  "props": Object {
                    "children": Array [
                      Object {
                        "uxpinPresetElementId": "2",
                      },
                    ],
                  },
                },
                "2": Object {
                  "name": "Card.Header",
                  "props": Object {
                    "children": Array [
                      Object {
                        "uxpinPresetElementId": "3",
                      },
                    ],
                    "title": "Card title",
                  },
                },
                "3": Object {
                  "name": "Card.Header.Button",
                  "props": Object {
                    "children": "Click me!",
                    "isPrimary": true,
                  },
                },
              },
              "name": "with direct imports",
              "rootId": "1",
            },
            Object {
              "elements": Object {
                "1": Object {
                  "name": "Card",
                  "props": Object {
                    "children": Array [
                      Object {
                        "uxpinPresetElementId": "2",
                      },
                    ],
                  },
                },
                "2": Object {
                  "name": "Card.Header",
                  "props": Object {
                    "children": Array [
                      Object {
                        "uxpinPresetElementId": "3",
                      },
                    ],
                    "title": "Card title",
                  },
                },
                "3": Object {
                  "name": "Card.Header.Button",
                  "props": Object {
                    "children": "Click me!",
                    "isPrimary": true,
                  },
                },
              },
              "name": "using namespaces",
              "rootId": "1",
            },
          ],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
          ],
          "wrappers": Array [],
        },
        Object {
          "defaultExported": true,
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Card/components/Header",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Card/components/Header/Header.tsx",
            },
            "presets": Array [],
          },
          "name": "Header",
          "namespace": Object {
            "importSlug": "Card_Header",
            "name": "Card",
          },
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": true,
              "name": "title",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
          ],
          "wrappers": Array [],
        },
        Object {
          "defaultExported": true,
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Card/components/Header/components/Button",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Card/components/Header/components/Button/Button.tsx",
            },
            "presets": Array [],
          },
          "name": "Button",
          "namespace": Object {
            "importSlug": "Card_Header_Button",
            "name": "Card.Header",
          },
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "isPrimary",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
          ],
          "wrappers": Array [],
        },
      ],
      "name": "Card",
    },
  ],
  "name": "Example Design System",
  "vcs": ObjectContaining {
    "branchName": Any<String>,
    "commitHash": StringMatching /\\[a-z0-9\\]\\+/,
  },
}
`;
